pkg/apiserver/utils/visual_plan.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
+
+package utils
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/golang/snappy"
+
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+// GenerateVisualPlan generate visual plan from raw data.
+func GenerateVisualPlan(v string) (*tipb.VisualData, error) {
+	if v == "" {
+		return nil, nil
+	}
+
+	// base64 decode
+	compressVPBytes, err := base64.StdEncoding.DecodeString(v)
+	if err != nil {
+		return nil, err
+	}
+
+	// snappy uncompress
+	vpBytes, err := snappy.Decode(nil, compressVPBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	// proto unmarshal
+	visual := &tipb.VisualData{}
+	err = visual.Unmarshal(vpBytes)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
It's wired that the impl of the compression algorithm and encoding algo are leaked from tidb.

Reply:
Why does tidb not provide the `tidb_decode_visual_plan` function as the design doc says?